webcomponents/tool/src/main/frontend/js/sakai-lti-popup.js

```diff
@@ -0,0 +1,69 @@
+import { SakaiElement } from "./sakai-element.js";
+import { html } from "./assets/lit-html/lit-html.js";
+
+class SakaiLTIPopup extends SakaiElement {
+  constructor() {
+    super();
+    const randomId = Math.floor(Math.random() * 1000000);
+    this.randomId = randomId;
+    this.preLaunchText = null;
+    this.postLaunchText = null;
+    this.loadTranslations("lti").then(t => {
+      this.i18n = t;
+      if ( this.preLaunchText == null ) this.preLaunchText = this.i18n.pre_launch_text;
+      if ( this.postLaunchText == null ) this.postLaunchText = this.i18n.post_launch_text;
+    } );
+  }
+
+  static get properties() {
+    return {
+      preLaunchText: {
+        attribute: "pre-launch-text",
+        type: String
+      },
+      postLaunchText: {
+        attribute: "post-launch-text",
+        type: String
+      },
+      launchUrl: {
+        attribute: "launch-url",
+        type: String
+      }
+    };
+  }
+
+  shouldUpdate() {
+    return this.preLaunchText && this.postLaunchText && this.launchUrl;
+  }
+
+  render() {
+    return html`
+<div>
+<button id="sakai-lti-popup-${this.randomId}" 
+    class="btn btn-primary" role="button"
+    onclick="
```

Comment:
You should use the @click handler. Just create a function in your class with all the stuff in the onclick handler, then add @click to the button tag. So, you'd add @click=${this.handlePopup} to the tag.